src/api/v3/inputs.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { faker } from '@faker-js/faker';
+import assert from 'assert';
+
+import { RecommendationsQueryVariables } from '../../generated/graphql/types';
+import {
+  GlobalRecsQueryParameterStrings,
+  handleQueryParameters,
+  setDefaultsAndCoerceTypes,
+} from './inputs';
+
+import { APIError, APIErrorResponse, BFFFxError } from '../../bfffxError';
+
+describe('input.ts recommendations query parameters', () => {
+  describe('setDefaultsAndCoerceTypes', () => {
+    it('converts count to an integer and passes through others', () => {
+      const res = setDefaultsAndCoerceTypes({
+        count: '3',
+        locale_lang: 'preValidatedLocale',
+        region: 'preValidatedRegion',
+      });
+      expect(res).toMatchObject({
+        count: 3,
+        locale: 'preValidatedLocale',
+        region: 'preValidatedRegion',
+      });
+    });
+
+    it('sets count to 20 if no default is provided, values without defaults are undefined', () => {
+      const res = setDefaultsAndCoerceTypes({});
+      // validation should return an error in this case, validating defaults though
+      expect(res).toMatchObject({
+        count: 20,
+      });
+    });
+  });
+
+  describe('handleQueryParameters', () => {
+    it('returns errors if invalid query parameters', () => {
+      const params: GlobalRecsQueryParameterStrings = {
+        count: '-1',
+      };
+
+      const error = handleQueryParameters(params);
+      assert(error instanceof BFFFxError);
+      const errors = JSON.parse(error.stringResponse);
+      expect(errors).toEqual(
+        expect.objectContaining<APIErrorResponse>({
+          errors: expect.arrayContaining<Array<APIError>>([
+            expect.objectContaining<APIError>({
+              status: '400',
+              title: 'Bad Request',
+            }),
+          ]),
+        })
+      );
+    });
```

Comment:
Nice! Wondering if we can use `toMatchObject` like above 🤔

Reply:
I copy-pasted these, so I'm not 100% sure, but I believe that `APIError` contains additional attributes that we didn't want to lock in tests? I don't know which one is better in this case.